File: plsql-frontend/src/test/java/org/sonar/plugins/plsqlopen/api/expressions/NewObjectExpressionTest.java
/*
 * Sonar PL/SQL Plugin (Community)
 * Copyright (C) 2015-2016 Felipe Zorzo
 * mailto:felipebzorzo AT gmail DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
package org.sonar.plugins.plsqlopen.api.expressions;

import static org.sonar.sslr.tests.Assertions.assertThat;

import org.junit.Before;
import org.junit.Test;
import org.sonar.plugins.plsqlopen.api.PlSqlGrammar;
import org.sonar.plugins.plsqlopen.api.RuleTest;

public class NewObjectExpressionTest extends RuleTest {

    @Before
    public void init() {
        setRootRule(PlSqlGrammar.EXPRESSION);
    }
    
    @Test
    public void matchesSimpleNewObject() {
        assertThat(p).matches("new foo.bar");
    }
    
    @Test
    public void matchesNewObjectWithEmptyParameterList() {
        assertThat(p).matches("new foo()");
    }
    
    @Test
    public void matchesNewObjectWithSimpleParameterList() {
        assertThat(p).matches("new foo(1, foo, 'bar', foo + bar)");
    }
    
    @Test
    public void matchesNewObjectWithNamedParameterList() {
        assertThat(p).matches("new foo(p1 => 1, p2 => foo, p3 => 'bar', p4 => foo + bar)");
    }
}

File: zpa-core/src/test/kotlin/com/felipebz/zpa/api/expressions/NewObjectExpressionTest.kt
/**
 * Z PL/SQL Analyzer
 * Copyright (C) 2015-2025 Felipe Zorzo
 * mailto:felipe AT felipezorzo DOT com DOT br
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
package com.felipebz.zpa.api.expressions

import com.felipebz.flr.tests.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import com.felipebz.zpa.api.PlSqlGrammar
import com.felipebz.zpa.api.RuleTest

class NewObjectExpressionTest : RuleTest() {

    @BeforeEach
    fun init() {
        setRootRule(PlSqlGrammar.EXPRESSION)
    }

    @Test
    fun matchesSimpleNewObject() {
        assertThat(p).matches("new foo.bar")
    }

    @Test
    fun matchesNewObjectWithEmptyParameterList() {
        assertThat(p).matches("new foo()")
    }

    @Test
    fun matchesNewObjectWithSimpleParameterList() {
        assertThat(p).matches("new foo(1, foo, 'bar', foo + bar)")
    }

    @Test
    fun matchesNewObjectWithNamedParameterList() {
        assertThat(p).matches("new foo(p1 => 1, p2 => foo, p3 => 'bar', p4 => foo + bar)")
    }
}
